datasync/chaindatafetcher/kafka/repository.go

```diff
@@ -75,6 +76,12 @@ func (r *repository) SetComponent(component interface{}) {
 func (r *repository) HandleChainEvent(event blockchain.ChainEvent, dataType types.RequestType) error {
 	switch dataType {
 	case types.RequestTypeBlockGroup:
+		if event.Block.NumberU64() > 0 {
+			err := checkStatesForSnapshot(r.blockchain, r.engine, event.Block.NumberU64()-1, event.Block.ParentHash())
+			if err != nil {
+				return err
```

Comment:
Return nil here because otherwise retryFunc will keep retrying this check

Reply:
Thanks. Applied at [f81391a](https://github.com/kaiachain/kaia/pull/48/commits/f81391ac662c24a91a00f92dffc5dc4c8bba3c10)